packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies.js

```diff
@@ -22,22 +30,52 @@ export function collectDependencies({ onlyNative, outputDir, widgetName }) {
             const resolvedPackagePath = await resolvePackage(
                 source,
                 dirname(importer ? importer : rollupOptions.input[0])
-            );
-            if (resolvedPackagePath && (!onlyNative || (await hasNativeCode(resolvedPackagePath)))) {
-                if (!managedDependencies.includes(resolvedPackagePath)) {
-                    managedDependencies.push(resolvedPackagePath);
+            ).catch(e => {
+                console.warn(e.message);
+                return null;
+            });
+
+            if (resolvedPackagePath) {
+                const checkOnlyNativeAndHasNativeCode = !onlyNative || (await hasNativeCode(resolvedPackagePath));
+                if (checkOnlyNativeAndHasNativeCode) {
+                    if (!managedDependencies.includes(resolvedPackagePath)) {
```

Comment:
I think it's nice to merge this nested condition with the parent into one condition. Thoughts?

Reply:
You're right. Done 👍